iso15118/shared/utils.py

```diff
@@ -1,10 +1,64 @@
 import asyncio
 import logging
-from typing import Coroutine, List
+from typing import Coroutine, List, Optional
+
+from iso15118.shared.exceptions import NoSupportedEnergyServices, NoSupportedProtocols
+from iso15118.shared.messages.enums import Protocol, ServiceV20
 
 logger = logging.getLogger(__name__)
 
 
+def _format_list(read_settings: List[str]) -> List[str]:
+    read_settings = list(filter(None, read_settings))
+    read_settings = [setting.strip().upper() for setting in read_settings]
+    read_settings = list(set(read_settings))
+    return read_settings
+
+
+def load_requested_protocols(read_protocols: Optional[List[str]]) -> List[Protocol]:
+    supported_protocols = [
+        "ISO_15118_2",
+        "ISO_15118_20_AC",
+        "DIN_SPEC_70121",
+    ]
+
+    protocols = _format_list(read_protocols)
+    valid_protocols = list(set(protocols).intersection(supported_protocols))
+    if not valid_protocols:
+        raise NoSupportedProtocols(
+            f"No supported protocols configured. Supported protocols are "
+            f"{supported_protocols} and could be configured in evcc_config.json"
+        )
+    supported_protocols = [Protocol[x] for x in valid_protocols]
+    return supported_protocols
```

Comment:
Same comment as above. The definition in secc_settings.py could be removed and use this one instead?

https://github.com/SwitchEV/iso15118/blob/2b146b483996863e82fca111db33dabf39f293fc/iso15118/secc/secc_settings.py#L126